### PR TITLE
Fix to build with HEAD

### PR DESCRIPTION
**Pull request by hsyl20**
Also remove some warnings

### CODE REVIEW

File: haddock-api/src/Haddock/Backends/Hyperlinker.hs [hunk 1]
```diff
@@ -18,7 +18,7 @@ import Data.Maybe
 import System.Directory
 import System.FilePath
 
-import GHC.Iface.Ext.Types  ( HieFile(..), HieASTs(..), HieAST(..), SourcedNodeInfo(..) )
+import GHC.Iface.Ext.Types
```
Comment (Kleidukos) [inline haddock-api/src/Haddock/Backends/Hyperlinker.hs:21]: Is the amount of stuff imported from Ext.Types too large to be explicitly listed?

Comment (hsyl20) [reply haddock-api/src/Haddock/Backends/Hyperlinker.hs:21]: I couldn't manage to import HiePath. It's just me not knowing how to import a pattern synonym ^^

Comment (Kleidukos) [reply haddock-api/src/Haddock/Backends/Hyperlinker.hs]: If I'm not mistaken: 

> Pattern synonyms can be exported and imported by prefixing the conid with the keyword pattern:

(_from https://gitlab.haskell.org/ghc/ghc/-/wikis/pattern-synonyms_)

Could you try this:

```suggestion
import GHC.Iface.Ext.Types (pattern HiePath, HieFile(..), HieASTs(..), HieAST(..), SourcedNodeInfo(..)) 
```

?

Comment (hsyl20) [reply haddock-api/src/Haddock/Backends/Hyperlinker.hs]: Ok now I know! Fixed

**Review by Kleidukos — APPROVED** [review]
LGTM, Godspeed!

**Comment by hsyl20** [comment]
Thanks @Kleidukos !

**Pull request merged**